src_c/scrap.c

```diff
@@ -355,6 +408,76 @@ _scrap_set_mode(PyObject *self, PyObject *args)
 
 #endif /* !defined(MAC_SCRAP) */
 
+/**
+ * @brief Fetches a python string from the SDL clipboard. If
+ *        there is nothing in the clipboard, it will return empty
+ *
+ * @return PyObject*
+ */
+static PyObject *
+_scrap_get_text(PyObject *self, PyObject *args)
+{
+    char *text = SDL_GetClipboardText();
+
+    if (!text) {
+        const char *err = SDL_GetError();
+        if (err) {
+            SDL_free(text);
+            PyErr_SetString(PyExc_RuntimeError, err);
+            return NULL;
+        }
```

Comment:
Hmmm-
so SDL_GetClipboardText says it sends back an empty string on failure, so is that the same as returning NULL? I don't think so? The char* would be a valid pointer to a 1 long array just containing NULL?

And then I have the same question about the point of `if (err)` like before.

Why is it a runtime error instead of a pygame.SDLerror?